connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -122,35 +160,56 @@ class UCSingleCatalog extends TableCatalog with SupportsNamespaces with Logging
 object UCSingleCatalog {
   val LOAD_DELTA_CATALOG = ThreadLocal.withInitial[Boolean](() => true)
   val DELTA_CATALOG_LOADED = ThreadLocal.withInitial[Boolean](() => false)
+
+  def generateCredentialProps(
+      scheme: String,
+      temporaryCredentials: TemporaryCredentials): Map[String, String] = {
+    if (scheme == "s3") {
+      val awsCredentials = temporaryCredentials.getAwsTempCredentials
+      Map(
+        // TODO: how to support s3:// properly?
+        "fs.s3a.access.key" -> awsCredentials.getAccessKeyId,
+        "fs.s3a.secret.key" -> awsCredentials.getSecretAccessKey,
+        "fs.s3a.session.token" -> awsCredentials.getSessionToken,
+        "fs.s3a.path.style.access" -> "true",
+        "fs.s3.impl.disable.cache" -> "true",
+        "fs.s3a.impl.disable.cache" -> "true"
+      )
+    } else if (scheme == "gs") {
+      val gcsCredentials = temporaryCredentials.getGcpOauthToken
+      Map(
+        GcsVendedTokenProvider.ACCESS_TOKEN_KEY -> gcsCredentials.getOauthToken,
+        GcsVendedTokenProvider.ACCESS_TOKEN_EXPIRATION_KEY -> temporaryCredentials.getExpirationTime.toString,
```

Comment:
@cloud-fan we need to add another configuration for GCS - `fs.gs.create.items.conflict.check.enable=false`

Reply:
Confirmed this locally - we could add the following to reduce the amount of client conf needed:
```scala
        "fs.gs.create.items.conflict.check.enable" -> "false",
        "fs.gs.auth.type" -> "ACCESS_TOKEN_PROVIDER",
        "fs.gs.auth.access.token.provider" -> classOf[GcsVendedTokenProvider].getName,
```

Reply:
done, thanks!